pkg/alizer/alizer.go

```diff
@@ -47,3 +53,63 @@ func GetDevfileLocationFromDetection(typ recognizer.DevFileType, registry api.Re
 		DevfileRegistry: registry.Name,
 	}
 }
+
+// DetectName retrieves the name of the project (if available)
+// If source code is detected:
+// 1. Detect the name (pom.xml for java, package.json for nodejs, etc.)
+// 2. If unable to detect the name, use the directory name
+//
+// If no source is detected:
+// 1. Use the directory name
+//
+// Last step. Sanitize the name so it's valid for a component name
+
+// Use:
+// import "github.com/redhat-developer/alizer/pkg/apis/recognizer"
+// components, err := recognizer.DetectComponents("./")
+
+// In order to detect the name, the name will first try to find out the name based on the program (pom.xml, etc.) but then if not, it will use the dir name.
+func DetectName(path string) (string, error) {
```

Comment:
Can you add a unit test for this function? I see we already have some tests in `pkg/alizer/alizer_test.go`..
I've been relying on this function in #6015 as a central place to retrieve component names, so I think it could be interesting to unit-test it.